src/lib/components/Viewer.tsx

```diff
@@ -36,19 +30,37 @@ export const Viewer = ({ data }: IViewerProps) => {
   // On every data change, convert it to binary and set it to the documentData state
   useEffect(() => {
     if (!data) return
-    const binary = base64ToBinary(data)
 
-    setDocumentData(binary)
+    // if data is string, convert it to binary
+    if (typeof data === 'string') {
```

Comment:
Binary can be encoded in a string too (our function `base64toBainary` returns Base64 binary in string), please check first if the data isnt in the right format first, if not, then do the conversion to Base64 Binary.